src/main/java/de/tum/in/www1/artemis/domain/statistics/CourseStatisticsAverageScore.java

```diff
@@ -9,17 +11,21 @@
 
     private final String exerciseName;
 
+    private final ZonedDateTime releaseDate;
+
     private double averageScore;
 
-    public CourseStatisticsAverageScore(long exerciseId, String exerciseName, double averageScore) {
+    public CourseStatisticsAverageScore(long exerciseId, String exerciseName, ZonedDateTime releaseDate, double averageScore) {
         this.exerciseId = exerciseId;
         this.exerciseName = exerciseName;
+        this.releaseDate = releaseDate;
         this.averageScore = averageScore;
     }
 
     public CourseStatisticsAverageScore() {
         this.exerciseId = 0L;
         this.exerciseName = "";
+        this.releaseDate = null;
```

Comment:
You don't have to assign `null` here since that will be the default value if nothing is assigned. (And removing this line also fixes the codacy warning.)

Reply:
not assigning null throws an error because releaseDate is final, so we have to set it somehow

Reply:
Ah, thanks for clarifying.